### DISCUSSION TITLE
Implement the unstable version of zwlr-screencopy-manager

### DISCUSSION THREAD

**Discussion by ThatOneCalculator** [YaLTeR/niri#464] [Ideas]
Niri has been great so far, but my biggest pain point is that I can't use https://github.com/russelltg/wl-screenrec -- it spits out `your compositor does not support zwlr-screencopy-manager and therefore is not support by wl-screenrec. See the README for supported compositors`

I saw that running the command in [this comment](https://github.com/YaLTeR/niri/discussions/232#discussioncomment-8610182) produces:

```
ext_idle_notifier_v1@v1
ext_session_lock_manager_v1@v1
org_kde_kwin_server_decoration_manager@v1
wl_compositor@v6
wl_data_device_manager@v3
wl_drm@v2
wl_output@v4
wl_seat@v9
wl_shm@v1
wl_subcompositor@v1
wp_cursor_shape_manager_v1@v1
wp_drm_lease_device_v1@v1
wp_fractional_scale_manager_v1@v1
wp_presentation@v1
wp_security_context_manager_v1@v1
wp_viewporter@v1
xdg_activation_v1@v1
xdg_wm_base@v6
zwlr_data_control_manager_v1@v2
zwlr_foreign_toplevel_manager_v1@v3
zwlr_gamma_control_manager_v1@v1
zwlr_layer_shell_v1@v4
zwlr_screencopy_manager_v1@v1
zwp_idle_inhibit_manager_v1@v1
zwp_input_method_manager_v2@v1
zwp_linux_dmabuf_v1@v5
zwp_pointer_constraints_v1@v1
zwp_pointer_gestures_v1@v3
zwp_primary_selection_device_manager_v1@v1
zwp_relative_pointer_manager_v1@v1
zwp_tablet_manager_v2@v1
zwp_text_input_manager_v3@v1
zwp_virtual_keyboard_manager_v1@v1
zxdg_decoration_manager_v1@v1
zxdg_exporter_v2@v1
zxdg_importer_v2@v1
zxdg_output_manager_v1@v3
```

in which I see both `zxdg_output_manager_v1` and `zwlr_screencopy_manager_v1@v1`, but it seems that wl-screenrec requires the *unstable* versions of those protocols.

**Comment by YaLTeR** [comment]
Yeah, I did not implement version 3 of the screencopy protocol because it's quite involved, while also being quite different from the existing portal screencast implementation. So maybe some day in the future; in the meantime I suggest something portal-based like OBS.

**Reply by ThatOneCalculator** [reply]
Gotcha. Is there anything else you could recommend besides OBS? A big draw of wl-screenrec is that it's super easy to just select an area of my screen to record.
